### PR TITLE
fix(client-certificates): return target errors on response when using http2

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts [hunk 1]
```diff
@@ -139,7 +146,6 @@ class SocksProxyConnection {
       dummyServer.emit('connection', this.internal);
       dummyServer.on('secureConnection', internalTLS => {
         debugLogger.log('client-certificates', `Browser->Proxy ${this.host}:${this.port} chooses ALPN ${internalTLS.alpnProtocol}`);
-        internalTLS.on('close', () => this.socksProxy._socksProxy.sendSocketEnd({ uid: this.uid }));
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts:142]: My assumption is that the 'close' event of the `tls.TLSSocket` will bubble down onto the `internal` `net.Socket`.

File: tests/library/client-certificates.spec.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ const { createHttpsServer, createHttp2Server } = require('../../packages/playwri
 type TestOptions = {
   startCCServer(options?: {
     http2?: boolean;
+    enableHTTP1FallbackWhenUsingHttp2?: boolean;
```
Comment (mxschmitt) [inline tests/library/client-certificates.spec.ts:28]: drive-by: more strict, enable it only in the test when we need it, line 308. This also makes the new test safer.

File: packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts [hunk 3]
```diff
@@ -169,20 +177,36 @@ class SocksProxyConnection {
         internalTLS.on('error', () => closeBothSockets());
         targetTLS.on('error', error => {
           debugLogger.log('client-certificates', `error when connecting to target: ${error.message}`);
+          const responseBody = 'Playwright client-certificate error: ' + error.message;
           if (internalTLS?.alpnProtocol === 'h2') {
-          // https://github.com/nodejs/node/issues/46152
-          // TODO: http2.performServerHandshake does not work here for some reason.
+            // This method is available only in Node.js 18+
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts:182]:
```suggestion
            // This method is available only in Node.js 20+
```

File: packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts [hunk 4]
```diff
@@ -169,20 +177,36 @@ class SocksProxyConnection {
...
+          const responseBody = 'Playwright client-certificate error: ' + error.message;
           if (internalTLS?.alpnProtocol === 'h2') {
-          // https://github.com/nodejs/node/issues/46152
-          // TODO: http2.performServerHandshake does not work here for some reason.
+            // This method is available only in Node.js 20+
+            if ('performServerHandshake' in http2) {
+              this._interceptClose = true;
+              // @ts-expect-error
+              const session = http2.performServerHandshake(internalTLS);
+              session.on('stream', (stream: http2.ServerHttp2Stream) => {
```
Comment (dgozman) [inline packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts:187]: Do we need an error handler on session/stream?

File: packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts [hunk 5]
```diff
@@ -169,20 +177,36 @@ class SocksProxyConnection {
         internalTLS.on('error', () => closeBothSockets());
         targetTLS.on('error', error => {
           debugLogger.log('client-certificates', `error when connecting to target: ${error.message}`);
+          const responseBody = 'Playwright client-certificate error: ' + error.message;
           if (internalTLS?.alpnProtocol === 'h2') {
-          // https://github.com/nodejs/node/issues/46152
-          // TODO: http2.performServerHandshake does not work here for some reason.
+            // This method is available only in Node.js 20+
+            if ('performServerHandshake' in http2) {
+              this._interceptClose = true;
```
Comment (dgozman) [inline packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts:184]: Can we instead remove the close listener, and move the respective comment here?

**Review by dgozman — APPROVED** [review]
